indexclient/client.py

```diff
@@ -385,6 +403,15 @@ def _doc_for_update(self):
     def _doc(self):
         return {k: self.__dict__[k] for k in self._attrs}
 
+    @property
+    def _sorted_doc(self):
+        """Return the _doc object but with all arrays in sorted order.
+
+        This will allow us to compare dictionaries with lists correctly. We
+        only care about the contents of the arrays not the order of them.
+        """
+        return recursive_sort(self._doc)
```

Comment:
could I suggest recursively converting the contents into sets, instead of sorting?

Reply:
Sets only work for hashable types in Python. So you can't have things like dictionaries inside of the lists.

```python
>>> { {1:1} }
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'dict'
```

Reply:
right but if we only care about the contents, only the lists are ordered right? so converting the lists to sets instead of sorting them should do

Reply:
I put a comment on the line I would change

Reply:
I guess you're right, this only works if the lists are lists only, and don't have more dictionaries inside them. I was thinking the indexd records will follow this but I'm not sure if there's something you might put in the metadata that would violate that

Reply:
Set is also not a hashable type, so the same is true for a list of lists, which would become a set of sets.

```python
>>> {{1}}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'set'

```

Reply:
it is actually possible using immutable sets :stuck_out_tongue: but probably not worth the trouble to care about the specific data types in the records...you should probably just ignore me haha